samples/azure/storage-persistence/ASP_2/StorageReader/AzureHelper.cs

```diff
@@ -31,45 +32,61 @@ public Task WriteOutData()
         var tableClient = storageAccount.CreateCloudBlobClient();
         var container = tableClient.GetContainerReference(containerName);
         Debug.WriteLine($"'{containerName}' container contents");
-        foreach (var blob in container.ListBlobs())
+        BlobContinuationToken token = null;
+        do
         {
-            var name = blob.Uri.AbsolutePath.Split('/').Last();
-            Debug.WriteLine($"  Blob:= {name}");
-            var blockBlobReference = container.GetBlockBlobReference(name);
-            var text = await blockBlobReference.DownloadTextAsync()
-                .ConfigureAwait(false);
-            Debug.WriteLine($"    {text}");
+            var segment = await container.ListBlobsSegmentedAsync(token);
+            token = segment.ContinuationToken;
+            foreach (var blob in segment.Results)
+            {
+                var name = blob.Uri.AbsolutePath.Split('/').Last();
+                Debug.WriteLine($"  Blob:= {name}");
+                var blockBlobReference = container.GetBlockBlobReference(name);
+                var text = await blockBlobReference.DownloadTextAsync()
+                    .ConfigureAwait(false);
+                Debug.WriteLine($"    {text}");
+            }
         }
+        while (token != null);
         Debug.WriteLine("");
     }
 
     #endregion
 
     #region WriteOutTable
 
-    static void WriteOutTable(string tableName, bool decodeRowKey)
+    static async Task WriteOutTable(string tableName, bool decodeRowKey)
     {
         var storageAccount = CloudStorageAccount.DevelopmentStorageAccount;
         var tableClient = storageAccount.CreateCloudTableClient();
         var table = tableClient.GetTableReference(tableName);
-        var entities = table.ExecuteQuery(new TableQuery()).ToList();
         Debug.WriteLine($"'{tableName}' table contents");
-        foreach (var entity in entities)
-        {
-            Debug.WriteLine($"  PartitionKey:= {entity.PartitionKey}");
-            Debug.WriteLine($"    RowKey:= {entity.RowKey}");
-            if (decodeRowKey)
-            {
-                var decodedRowKey = entity.RowKey.DecodeFromKey();
-                Debug.WriteLine($"    DecodedRowKey:= {decodedRowKey}");
-            }
-            foreach (var property in entity.Properties)
+        TableContinuationToken token = null;
+        do {
```

Comment:
Not sure if this formatting is correct. Can you reformat this?

Reply:
done